src/object/obj_rpc.c

```diff
@@ -122,67 +82,52 @@ crt_proc_daos_epoch_range_t(crt_proc_t proc, daos_epoch_range_t *erange)
 	int rc;
 
 	rc = crt_proc_uint64_t(proc, &erange->epr_lo);
-	if (rc != 0)
-		return -DER_HG;
+	if (unlikely(rc))
+		return rc;
 
 	rc = crt_proc_uint64_t(proc, &erange->epr_hi);
-	if (rc != 0)
-		return -DER_HG;
-
-	return 0;
-}
-
+	if (unlikely(rc))
+		return rc;
 
-static int
-crt_proc_struct_obj_shard_iod(crt_proc_t proc, struct obj_shard_iod *siod)
-{
-	if (crt_proc_uint32_t(proc, &siod->siod_tgt_idx) != 0)
-		return -DER_HG;
-	if (crt_proc_uint32_t(proc, &siod->siod_idx) != 0)
-		return -DER_HG;
-	if (crt_proc_uint32_t(proc, &siod->siod_nr) != 0)
-		return -DER_HG;
-	if (crt_proc_uint64_t(proc, &siod->siod_off) != 0)
-		return -DER_HG;
 	return 0;
 }
 
 static int
 crt_proc_struct_obj_io_desc(crt_proc_t proc, struct obj_io_desc *oiod)
 {
 	crt_proc_op_t	proc_op;
-	uint32_t	i;
 	int		rc;
 
 	rc = crt_proc_uint16_t(proc, &oiod->oiod_nr);
-	if (rc)
-		return -DER_HG;
+	if (unlikely(rc))
+		return rc;
 	rc = crt_proc_uint16_t(proc, &oiod->oiod_tgt_idx);
-	if (rc)
-		return -DER_HG;
+	if (unlikely(rc))
+		return rc;
 	rc = crt_proc_uint32_t(proc, &oiod->oiod_flags);
-	if (rc)
-		return -DER_HG;
+	if (unlikely(rc))
+		return rc;
 
 	rc = crt_proc_get_op(proc, &proc_op);
-	if (rc != 0)
-		return -DER_HG;
-	if (proc_op == CRT_PROC_DECODE && oiod->oiod_nr > 0) {
+	if (unlikely(rc))
+		return rc;
+	if (DECODING(proc_op) && oiod->oiod_nr > 0) {
 		rc = obj_io_desc_init(oiod, oiod->oiod_nr, oiod->oiod_flags);
-		if (rc)
+		if (unlikely(rc))
 			return rc;
 	}
 
-	for (i = 0; oiod->oiod_siods != NULL && i < oiod->oiod_nr; i++) {
-		rc = crt_proc_struct_obj_shard_iod(proc, &oiod->oiod_siods[i]);
-		if (rc != 0) {
-			if (proc_op == CRT_PROC_DECODE)
+	if (oiod->oiod_siods != NULL) {
+		rc = crt_proc_memcpy(proc, oiod->oiod_siods,
+				   oiod->oiod_nr * sizeof(oiod->oiod_siods[0]));
```

Comment:
```suggestion
				     oiod->oiod_nr * sizeof(oiod->oiod_siods[0]));
```

Reply:
Cannot do this because of out of 80 chars in line.